lib/src/captions/cea608-708/CaptionScreen.ts

```diff
@@ -0,0 +1,253 @@
+		row.moveCursor(relPos);
+	}
+
+	setCursor(absPos: number) {
+		this.logger.log(VerboseLevel.INFO, 'setCursor: ' + absPos);
+		const row = this.rows[this.currRow];
+		row.setCursor(absPos);
+	}
+
+	setPAC(pacData: PACData) {
+		this.logger.log(
+			VerboseLevel.INFO,
+			() => 'pacData = ' + JSON.stringify(pacData),
+		);
+		let newRow = pacData.row - 1;
+		if (this.nrRollUpRows && newRow < this.nrRollUpRows - 1) {
+			newRow = this.nrRollUpRows - 1;
+		}
+
+		// Make sure this only affects Roll-up Captions by checking this.nrRollUpRows
+		if (this.nrRollUpRows && this.currRow !== newRow) {
+			// clear all rows first
+			for (let i = 0; i < NR_ROWS; i++) {
+				this.rows[i].clear();
+			}
+
+			// Copy this.nrRollUpRows rows from lastOutputScreen and place it in the newRow location
+			// topRowIndex - the start of rows to copy (inclusive index)
+			const topRowIndex = this.currRow + 1 - this.nrRollUpRows;
+			// We only copy if the last position was already shown.
+			// We use the cueStartTime value to check this.
+			const lastOutputScreen = this.lastOutputScreen;
+			if (lastOutputScreen) {
+				const prevLineTime = lastOutputScreen.rows[topRowIndex].cueStartTime;
+				const time = this.logger.time;
+				if (prevLineTime !== null && time !== null && prevLineTime < time) {
+					for (let i = 0; i < this.nrRollUpRows; i++) {
+						this.rows[newRow - this.nrRollUpRows + i + 1].copy(
+							lastOutputScreen.rows[topRowIndex + i],
+						);
+					}
+				}
+			}
+		}
+
+		this.currRow = newRow;
+		const row = this.rows[this.currRow];
+		if (pacData.indent !== null) {
+			const indent = pacData.indent;
+			const prevPos = Math.max(indent - 1, 0);
+			row.setCursor(pacData.indent);
+			pacData.color = row.chars[prevPos].penState.foreground;
+		}
+		const styles: PenStyles = {
+			foreground: pacData.color,
+			underline: pacData.underline,
+			italics: pacData.italics,
+			background: 'black',
+			flash: false,
+		};
+		this.setPen(styles);
+	}
+
+	/**
+   * Set background/extra foreground, but first do back_space, and then insert space (backwards compatibility).
+   */
+	setBkgData(bkgData: Partial<PenStyles>) {
+		this.logger.log(
+			VerboseLevel.INFO,
+			() => 'bkgData = ' + JSON.stringify(bkgData),
+		);
+		this.backSpace();
+		this.setPen(bkgData);
+		this.insertChar(0x20); // Space
+	}
+
+	setRollUpRows(nrRows: number | null) {
+		this.nrRollUpRows = nrRows;
+	}
+
+	rollUp() {
+		if (this.nrRollUpRows === null) {
+			this.logger.log(
+				VerboseLevel.DEBUG,
+				'roll_up but nrRollUpRows not set yet',
+			);
+			return; // Not properly setup
+		}
+		this.logger.log(VerboseLevel.TEXT, () => this.getDisplayText());
+		const topRowIndex = this.currRow + 1 - this.nrRollUpRows;
+		const topRow = this.rows.splice(topRowIndex, 1)[0];
+		topRow.clear();
+		this.rows.splice(this.currRow, 0, topRow);
+		this.logger.log(VerboseLevel.INFO, 'Rolling up');
+		// this.logger.log(VerboseLevel.TEXT, this.get_display_text())
```

Comment:
Does this commented log fulfil any purpose if de-commented?

Reply:
I de-commented the line and tried it as it is and didnt work, on my opinion I would consider this zombie code and just erase it